logstash-core/lib/logstash/pipeline.rb

```diff
@@ -34,6 +34,7 @@ module LogStash; class BasePipeline
 
   def initialize(config_str, settings = SETTINGS)
     @logger = self.logger
+    @mutex = Mutex.new
```

Comment:
I didn't want to imply that the mutex was only to be used for the write client. I think it can be general purpose. I was actually a bit surprised that one didn't already exist for this class, had it, i would have just used that.